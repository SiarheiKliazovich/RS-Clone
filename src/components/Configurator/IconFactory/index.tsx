import { FunctionComponent } from "react";
import { IconFactoryType } from "../../../types/configurator";
import { IIconMapping } from "../../../interfaces/configurator";

const IconFactory: FunctionComponent<IconFactoryType> = ({
  components,
  settings,
  type,
}: IconFactoryType) => {
  /* eslint-disable @typescript-eslint/no-non-null-assertion */
  const IconComponent = components.find(
    (component: IIconMapping) => type === component.type
  )!.component;

  return <IconComponent settings={settings} />;
};
export default IconFactory;
