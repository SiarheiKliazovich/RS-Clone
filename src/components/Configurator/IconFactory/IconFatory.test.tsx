import { render } from "@testing-library/react";
import IconFactory from "../IconFactory";

describe("IconFactory", () => {
  const components = [
    {
      type: "type1",
      component: () => <div>Icon 1</div>,
    },
    {
      type: "type2",
      component: () => <div>Icon 2</div>,
    },
  ];

  const settings = {
    color: "#ff0000",
    size: "16px",
    weave: "weave",
    bgColor: "#000",
    colorOne: "#888",
    colorTwo: "#888",
    colorThree: "#333",
    colorFour: "#444",
    colorFive: "#111",
    colorSix: "#888",
  };

  test("renders correctly", () => {
    const { container } = render(
      <IconFactory components={components} type="type1" settings={settings} />
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
