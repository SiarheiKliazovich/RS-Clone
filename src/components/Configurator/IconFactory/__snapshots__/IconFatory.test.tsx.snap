// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconFactory renders correctly 1`] = `
<div>
  Icon 1
</div>
`;
